graylog2-server/src/main/java/org/graylog2/entitygroups/rest/EntityGroupResource.java

```diff
@@ -0,0 +1,178 @@
+@RequiresAuthentication
+@Produces(MediaType.APPLICATION_JSON)
+public class EntityGroupResource extends RestResource implements PluginRestResource {
+
+    private final EntityGroupService entityGroupService;
+
+    @Inject
+    public EntityGroupResource(EntityGroupService entityGroupService) {
+        this.entityGroupService = entityGroupService;
+    }
+
+    @GET
+    @ApiOperation(value = "Get a list of entity groups")
+    @RequiresPermissions(EntityGroupPermissions.ENTITY_GROUP_READ)
+    public PaginatedList<EntityGroup> listGroups(@ApiParam(name = "page") @QueryParam("page") @DefaultValue("1") int page,
+                                                 @ApiParam(name = "per_page") @QueryParam("per_page") @DefaultValue("15") int perPage,
+                                                 @ApiParam(name = "query") @QueryParam("query") @DefaultValue("") String query,
+                                                 @ApiParam(name = "sort",
+                                                           value = "The field to sort the result on",
+                                                           allowableValues = "name")
+                                                 @QueryParam("sort") String sort,
+                                                 @ApiParam(name = "direction", value = "The sort direction", allowableValues = "asc,desc")
+                                                 @DefaultValue("asc") @QueryParam("direction") SortOrder order) {
+        return entityGroupService.findPaginated(query, page, perPage, order, sort, null);
+    }
+
+    @GET
+    @Path("/{id}")
+    @ApiOperation(value = "Get a single entity group")
+    @ApiResponses(@ApiResponse(code = 404, message = "No such entity group"))
+    @RequiresPermissions(EntityGroupPermissions.ENTITY_GROUP_READ)
+    public Response get(@ApiParam(name = "id", required = true) @PathParam("id") String id) {
+        try {
+            return Response.ok().entity(entityGroupService.requireEntityGroup(id)).build();
+        } catch (IllegalArgumentException e) {
+            throw new BadRequestException(getRootCauseMessage(e), e);
+        }
+    }
+
+    @GET
+    @Path("/get_for_entity")
+    @ApiOperation(value = "Get a list of entity groups for an entity")
+    @RequiresPermissions(EntityGroupPermissions.ENTITY_GROUP_READ)
+    public Response getAllForEntity(@ApiParam(name = "entity_type", required = true) @QueryParam("entity_type") String entityType,
+                                    @ApiParam(name = "entity_id", required = true) @QueryParam("entity_id") String entityId) {
+        return Response.ok().entity(entityGroupService.getAllForEntity(entityType, entityId)).build();
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @ApiOperation("Create a new entity group")
+    @ApiResponses(@ApiResponse(code = 400, message = "An entity group already exists with id or name"))
+    @RequiresPermissions(EntityGroupPermissions.ENTITY_GROUP_EDIT)
```

Comment:
This is fine as an annotation, because there is no instance to check yet, but this should use a "create" permission instead of "edit".